### PR TITLE
[GNA] Fixed export/import functionality

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
it fixes the root cause of export/import bugs: using nGroup to guess input/output dimensions.
1) Extended RuntimeEndPoint structure by including information about input/output dims, precision and layout.
2) Add the new import header (v2.7) and logic to import the old versions of the exported models
3) Enabled versioning functionality for RuntimeEndPoint  like it was done for ModelHeader

### Tickets:
 - 55662
 - 51527

### CODE REVIEW

**Review by elilobanova — COMMENTED** [review]
Please, add some tests or maybe update existed for import/export problems.

**Comment by nosovmik** [comment]
Can you remove appropriate tests from 'skip_test_config' for issues 51527 and 51528?

File: inference-engine/src/gna_plugin/gna_model_serial.cpp [hunk 1]
```diff
@@ -796,13 +834,22 @@ std::vector<HeaderLatest::RuntimeEndPoint> GNAModelSerial::serializeOutputs(cons
     std::size_t outputIndex = 0;
     for (auto const &output : outputsDataMap) {
         auto outputName = output.first;
-        auto inputDims = output.second->getTensorDesc().getDims();
-        uint32_t elementsCount = static_cast<uint32_t>(InferenceEngine::details::product(inputDims.begin(), inputDims.end()));
-
+        auto outputDims = output.second->getTensorDesc().getDims();
+        struct Gna2Shape outputShape = {0, {0}};
+        outputShape.NumberOfDimensions = outputDims.size();
+        for ( size_t i=0; i < outputShape.NumberOfDimensions; ++i) {
```
[the remark this answers is not in the document]
Comment (mryzhov) [reply inference-engine/src/gna_plugin/gna_model_serial.cpp]: fixed

File: inference-engine/src/gna_plugin/gna_model_serial.cpp [hunk 2]
```diff
@@ -846,20 +901,25 @@ void GNAModelSerial::ImportInputs(std::istream &is,
...
+
+        HeaderLatest::RuntimeEndPoint input = ReadEndPoint(is);
         inputsDesc->getPtrInputsGlobal(name).push_back(reinterpret_cast<float*>(reinterpret_cast<uint8_t *> (basePtr) + input.descriptor_offset));
         inputsDesc->orientation_in[name] = input.orientation;
         inputsDesc->bytes_allocated_for_input[name] = input.element_size * input.elements_count;
 
-        auto inputDims = InferenceEngine::SizeVector({modelHeader.nGroup, input.elements_count / modelHeader.nGroup});
-
+        auto inputDims = InferenceEngine::SizeVector();
+        for (auto i = 0; i < input.shape.NumberOfDimensions; ++i)
```
[the remark this answers is not in the document]
Comment (mryzhov) [reply inference-engine/src/gna_plugin/gna_model_serial.cpp]: fixed

**Review by nkogteva — APPROVED** [review]

File: inference-engine/src/gna_plugin/serial/headers/2dot7/gna_model_header.hpp [hunk 3]
```diff
@@ -0,0 +1,181 @@
+// Copyright (C) 2018-2021 Intel Corporation
```
Comment (elilobanova) [inline inference-engine/src/gna_plugin/serial/headers/2dot7/gna_model_header.hpp:1]: 2018 seems to be extra in a new file

**Comment by mryzhov** [comment]
> Can you remove appropriate tests from 'skip_test_config' for issues 51527 and 51528?
I tried and the initial issues have been gone, but the tests still fail during the network loading because of GNA limitation.
So I suggest to prepare additional PR for the tests

**Review by elilobanova — APPROVED** [review]

File: inference-engine/src/gna_plugin/serial/headers/2dot7/gna_model_header.hpp (hunk 3 above)
Comment (mryzhov) [reply inference-engine/src/gna_plugin/serial/headers/2dot7/gna_model_header.hpp]: fixed

**Review by azhogov — APPROVED** [review]

**Pull request merged**